### PR TITLE
cache.go minor refactoring

### PR DESCRIPTION
**Pull request by WindzCUHK**
# Description

- remove package dependency

## Type of change

- [ ] Bug fix
- [ ] New feature
- [x] Refactoring (no functional changes, no api changes)
- [ ] Non-code changes (update documentation, pipeline, etc.)

### Flags

- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code

## Related issue/PR

- from: https://github.com/AthenZ/k8s-athenz-sia/pull/103

---

## Checklist

- [x] Followed the guidelines in the CONTRIBUTING document
- [x] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [x] Tested and linted the code
- [x] Commented the code
- [x] Made corresponding changes to the documentation
- [x] Passed all pipeline checking

## Checklist for maintainer

- Use `Squash and merge`
- Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- Delete the branch after merge

### CODE REVIEW

File: pkg/token/cache.go [hunk 1]
```diff
@@ -84,13 +83,13 @@ type LockedTokenCache struct {
 	podName string
 }
 
-func NewLockedTokenCache(tokenType string, idConfig *config.IdentityConfig) *LockedTokenCache {
+func NewLockedTokenCache(tokenType, namespace, podName string) *LockedTokenCache {
```
Comment (mlajkim) [inline pkg/token/cache.go:86]: Why would you not use the pointer?
- easy to read
- less parameters
- faster

**Review by mlajkim — COMMENTED** [review]
.

File: pkg/token/cache.go (hunk 1 above)
Comment (WindzCUHK) [reply pkg/token/cache.go:86]: > Why would you not use the pointer?
- remove the dependency from `cache.go` to `config.go`, keep them decoupled
- easy to read => do not know which field will be used
- less parameters => should keep to interface to minimum (IdentityConfig is not 1 object, it implies all its fields)
- faster => agress

Comment (mlajkim) [reply pkg/token/cache.go:86]: > remove the dependency from cache.go to config.go, keep them decoupled

okay

**Review by mlajkim — APPROVED** [review]

**Pull request merged**